api/v1alpha1/api.go

```diff
@@ -144,3 +166,87 @@ const (
 	// This can be used to describe the routing behavior in HTTPRoute referenced by LLMRoute.
 	LLMModelHeaderKey = "x-envoy-ai-gateway-llm-model"
 )
+
+// LLMProviderAuthenticationType specifies the type of auth mechanism used to access a Provider.
+type LLMProviderAuthenticationType string
+
+const (
+	LLMProviderAuthenticationTypeAPIKey    LLMProviderAuthenticationType = "APIKey"
+	LLMProviderAuthenticationTypeOIDC      LLMProviderAuthenticationType = "OIDC"
+	LLMProviderAuthenticationTypeStaticKey LLMProviderAuthenticationType = "StaticKey"
+)
+
+// +kubebuilder:object:root=true
+
+// BackendSecurityPolicy specifies configuration for authentication and authorization rules on the traffic
+// exiting the gateway to the backend.
+type BackendSecurityPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              BackendSecurityPolicySpec `json:"spec,omitempty"`
+}
+
+// BackendSecurityPolicySpec specifies authentication and authorization rules on access the provider from the Gateway.
```

Comment:
The authorization rules are checked from client to gateway, gateway to LLM provider is only authentication.

Reply:
Won't the API key be used to check authorization in addition to authentication ie using AWS credentials for the gateway to access the LLM?